src/dali/abstract_ccxt_pair_converter_plugin.py

```diff
@@ -513,16 +514,23 @@ def find_historical_bar(self, from_asset: str, to_asset: str, timestamp: datetim
     def find_historical_bars(
         self, from_asset: str, to_asset: str, timestamp: datetime, exchange: str, all_bars: bool = False, timespan: str = _MINUTE
     ) -> Optional[List[HistoricalBar]]:
+
+        # Guard clause to pull from cache
+        if all_bars:
+            cached_bundle = self._get_bundle_from_cache(AssetPairAndTimestamp(timestamp, from_asset, to_asset, exchange))
+            if cached_bundle:
+                # If the last bar in the bundle is within the last week, return the bundle
+                if (datetime.now(timezone.utc) - cached_bundle[-1].timestamp).total_seconds() <= _TIME_GRANULARITY_STRING_TO_SECONDS[_ONE_WEEK]:
```

Comment:
@eprbell I don't think I can use a set time for the example bars, because this guard clause checks for stale data by checking if the stamp is within the last week.